clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonApplicationLoadBalancerCachingAgent.groovy

```diff
@@ -38,21 +47,39 @@ import com.netflix.spinnaker.clouddriver.aws.AmazonCloudProvider
 import com.netflix.spinnaker.clouddriver.aws.data.ArnUtils
 import com.netflix.spinnaker.clouddriver.aws.data.Keys
 import com.netflix.spinnaker.clouddriver.aws.edda.EddaApi
+import com.netflix.spinnaker.clouddriver.aws.model.InstanceTargetGroupState
+import com.netflix.spinnaker.clouddriver.aws.model.InstanceTargetGroups
 import com.netflix.spinnaker.clouddriver.aws.model.edda.EddaRule
+import com.netflix.spinnaker.clouddriver.aws.model.edda.TargetGroupAttributes
+import com.netflix.spinnaker.clouddriver.aws.model.edda.TargetGroupHealth
 import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider
 import com.netflix.spinnaker.clouddriver.aws.security.EddaTimeoutConfig
 import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials
 import com.netflix.spinnaker.clouddriver.cache.OnDemandAgent
+import com.netflix.spinnaker.clouddriver.core.provider.agent.HealthProvidingCachingAgent
 import retrofit.RetrofitError
 
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.INFORMATIVE
+import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.HEALTH
+import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.INSTANCES
 import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.LOAD_BALANCERS
 import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.ON_DEMAND
 import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.TARGET_GROUPS
 
-class AmazonApplicationLoadBalancerCachingAgent extends AbstractAmazonLoadBalancerCachingAgent {
+class AmazonApplicationLoadBalancerCachingAgent extends AbstractAmazonLoadBalancerCachingAgent implements HealthProvidingCachingAgent {
   final EddaApi eddaApi
   final EddaTimeoutConfig eddaTimeoutConfig
 
+  static final Collection<AgentDataType> types = Collections.unmodifiableCollection([
+    AUTHORITATIVE.forType(LOAD_BALANCERS.ns),
+    AUTHORITATIVE.forType(TARGET_GROUPS.ns),
+    AUTHORITATIVE.forType(HEALTH.ns),
+    INFORMATIVE.forType(INSTANCES.ns)
+  ])
+
+  static final String healthId = "aws-load-balancer-v2-target-group-instance-health"
```

Comment:
maybe `private static final String HEALTH_ID` or something ... it looks too much like a regular class variable.